src/main/groovy/net/vivin/gradle/versioning/tasks/TagTask.groovy

```diff
@@ -9,12 +9,23 @@ import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.gradle.tooling.BuildException
 
+import java.util.concurrent.Callable
+
 /**
  * Created on 6/23/16 at 10:26 AM
  * @author vivin
  */
 class TagTask extends DefaultTask {
     @Internal
+    Callable<?> message = { '' }
+
+    private Closure<String> fromEnvironmentVariableClosure = { System.env."$it" }
+
+    private Closure<String> fromSystemPropertyClosure = { System.properties."$it" }
+
+    private Closure<Object> fromProjectPropertyClosure = { project.hasProperty(it) ? project.property(it) : null }
```

Comment:
One thing I was thinking. What if you use this and you don't set the property? Do you think it would be worth it to throw an error if the property doesn't exist? Just to prevent the situation where they don't provide one accidentally and we end up creating a lightweight tag instead? Maybe we can have a "strict" version and a non-strict version of the closure?